Basic date/time support

benesch opened the pull request:
This adds support for parsing date/time types as string literals. It doesn't try to make sense of the data inside the string literals by e.g. returning chrono objects; that's left to downstream libraries for now.

Note that this conflicts with #90 and #96. To reduce the merge conflicts, I've simply based this PR off the merge of those two PRs.

benesch commented:
Oh, I guess this is #62.

nickolay commented:
The DATE/TIME/TIMESTAMP commit looks good to me, though I wonder why you chose to deviate from the SCREAMING CASE we use for SQL code in this repo. (At work I prefer lower-case too, but we've got to value consistency.)

The INTERVAL commit (and the remaining PRs) is more complex, and I'm not sure when I'll have the time to give it the attention it deserves. Please feel free to merge any of them without waiting for me, if it makes it any easier for you.

benesch commented:
> The DATE/TIME/TIMESTAMP commit looks good to me, though I wonder why you chose to deviate from the SCREAMING CASE we use for SQL code in this repo. (At work I prefer lower-case too, but we've got to value consistency.)

Oh, shoot, I fixed that oversight but amended it into the wrong commit. Updated!

> The INTERVAL commit (and the remaining PRs) is more complex, and I'm not sure when I'll have the time to give it the attention it deserves. Please feel free to merge any of them without waiting for me, if it makes it any easier for you.

No worries at all! I'll extract it into its own PR so you can review it at your leisure, and merge this PR in the meantime.
